### ISSUE TITLE
Function Suggestion Encode

### ISSUE THREAD

**Issue by lucas19961** [MrBrax/LiveStreamDVR#272]
Function to Encode the VODS to get smaller file sizes

**Comment by MrBrax** [comment]
seems very obvious yeah, would need some very finely tuned presets for it to work well

**Comment by MrBrax** [comment]
reencoder added to develop branch, but unless you know what combos of options to use, it might just instantly crash

**Comment by Serph91P** [comment]
Yeah tested it with one stream, crashed the whole application. Passed through my GPU to enable NVENC HVEC encode. It says that it would use a preset? Log is attached dunno what else to setup so it wont crash.
[crash_log.txt](https://github.com/MrBrax/LiveStreamDVR/files/9475118/crash_log.txt)

**Comment by MrBrax** [comment]
for the actual crash i might fix that, but the cause might be found in the software folder regarding ffmpeg

**Comment by Serph91P** [comment]
> for the actual crash i might fix that, but the cause might be found in the software folder regarding ffmpeg

Found these two logs.
[reencode_h0llylp_202209_2.mp4_stderr.log](https://github.com/MrBrax/LiveStreamDVR/files/9476153/reencode_h0llylp_202209_2.mp4_stderr.log)
[reencode_h0llylp_202209_2.mp4_stdout.log](https://github.com/MrBrax/LiveStreamDVR/files/9476154/reencode_h0llylp_202209_2.mp4_stdout.log)

**Comment by MrBrax** [comment]
so it can't find the nvenc codec, probably tells me that the docker ffmpeg doesn't support hardware acceleration out of the box just like that

**Comment by Serph91P** [comment]
> so it can't find the nvenc codec, probably tells me that the docker ffmpeg doesn't support hardware acceleration out of the box just like that

Just read the log myself, yeah it just states hvec:nvenc unknown. First i thought maybe my gpu does not support it but it should since its working in plex, transocding

**Comment by Serph91P** [comment]
> so it can't find the nvenc codec, probably tells me that the docker ffmpeg doesn't support hardware acceleration out of the box just like that

found this on the web about it dunno about the effort

**Comment by MrBrax** [comment]
yes the problem is that i'd need to change the base docker image adding probably tons of bloat, probably a bit out of scope

**Comment by Serph91P** [comment]
> yes the problem is that i'd need to change the base docker image adding probably tons of bloat, probably a bit out of scope

dunno if this will help not so deep into linux
https://github.com/vexorian/dizquetv/issues/218

**Comment by MrBrax** [comment]
it would, but then it would not build from the npm image leaving the most crucial part out. dunno if two images can be merged, haven't studied docker much

**Comment by Serph91P** [comment]
> it would, but then it would not build from the npm image leaving the most crucial part out. dunno if two images can be merged, haven't studied docker much

its done here no?
https://github.com/vexorian/dizquetv/blob/main/Dockerfile-nvidia

**Comment by MrBrax** [comment]
interesting, might have a look into it later

**Comment by Serph91P** [comment]
> interesting, might have a look into it later

shouldnt this also work in docker?
https://docs.nvidia.com/video-technologies/video-codec-sdk/ffmpeg-with-nvidia-gpu/

**Comment by MrBrax** [comment]
it probably would, but there's a lot of debugging and all that, i don't run linux so i can't test if it even works, and it would add a lot of bloat for people not using it
